src/lib.rs

```diff
@@ -51,16 +91,64 @@ mod sys;
 #[path = "linux.rs"]
 mod sys;
 
-#[cfg(target_env = "sgx")]
-#[path = "sgx.rs"]
-mod sys;
+#[cfg(not(any(target_os = "linux", target_os = "macos", target_arch = "wasm32")))]
+pub struct Platform {
+    _priv: (),
+}
+
+#[cfg(not(any(target_os = "linux", target_os = "macos", target_arch = "wasm32")))]
+impl Platform {
+    const fn new() -> Platform {
+        Platform { _priv: () }
+    }
+}
+
+#[cfg(not(any(target_os = "linux", target_os = "macos", target_arch = "wasm32")))]
+unsafe impl System for Platform {
+    fn alloc(&self, size: usize) -> (*mut u8, usize, u32) {
+        panic!("Please implement the `System` interface and use the `Dlmalloc::new_with_platform` method");
+    }
+
+    fn remap(&self, ptr: *mut u8, oldsize: usize, newsize: usize, can_move: bool) -> *mut u8 {
+        panic!("Please implement the `System` interface and use the `Dlmalloc::new_with_platform` method");
+    }
 
-impl Dlmalloc {
-    /// Creates a new instance of an allocator, same as `DLMALLOC_INIT`.
-    pub fn new() -> Dlmalloc {
-        DLMALLOC_INIT
+    fn free_part(&self, ptr: *mut u8, oldsize: usize, newsize: usize) -> bool {
+        panic!("Please implement the `System` interface and use the `Dlmalloc::new_with_platform` method");
     }
 
+    fn free(&self, ptr: *mut u8, size: usize) -> bool {
+        panic!("Please implement the `System` interface and use the `Dlmalloc::new_with_platform` method");
+    }
+
+    fn can_release_part(&self, flags: u32) -> bool {
+        panic!("Please implement the `System` interface and use the `Dlmalloc::new_with_platform` method");
+    }
+
+    fn allocates_zeros(&self) -> bool {
+        panic!("Please implement the `System` interface and use the `Dlmalloc::new_with_platform` method");
+    }
+
+    fn page_size(&self) -> usize {
+        panic!("Please implement the `System` interface and use the `Dlmalloc::new_with_platform` method");
+    }
+}
+
+impl Dlmalloc<Platform> {
+    /// Creates a new instance of an allocator
+    pub const fn new() -> Dlmalloc<Platform> {
+        Dlmalloc(dlmalloc::Dlmalloc::new(Platform::new()))
+    }
+}
+
+impl<S> Dlmalloc<S> {
+    /// Creates a new instance of an allocator
+    pub const fn new_with_platform(sys_allocator: S) -> Dlmalloc<S> {
```

Comment:
I think this may want to be `new_with_system` since the name of the trait is "System"?

Reply:
This can also be lumped in with the `impl` block below I think.

Reply:
Unfortunately, that results in `trait bounds other than `Sized` on const fn parameters are unstable`. I'm assuming you'd want to avoid a `const_fn` crate attribute?